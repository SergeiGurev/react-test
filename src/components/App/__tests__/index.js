import React from 'react'
import { shallow } from 'enzyme'
import { shallowToJson } from 'enzyme-to-json'

import { App } from '..'

const props = {
  store: {
    countries: [
      {
        key: 'ru',
        value: 'ru',
        text: 'Russia'
      },
      {
        key: 'ua',
        value: 'ua',
        text: 'Ukraine'
      }
    ],
    isCountriesLoading: false,
    selectedCountry: 'ru',
    cities: [
      {
        country: 'Russia',
        latitude: '43',
        longitude: '42',
        region: 'Leningradskaya',
        title: 'Saint-Peterspurg',
      },
      {
        country: 'Ukraine',
        latitude: '42',
        longitude: '43',
        region: 'Kievskaia',
        title: 'Kiev',
      }
    ],
    selectedCity: 'Saint-Peterspurg',
    temperature: 26,
    isSearchLoading: false,
    searchResults: [],
    searchValue: '',
    isWeatherLoading: false,
  }
}

describe('App component', () => {
  it('renders without error', () => {
    const tree = shallow(<App {...props} />)
    expect(shallowToJson(tree)).toMatchSnapshot()
  })
})
