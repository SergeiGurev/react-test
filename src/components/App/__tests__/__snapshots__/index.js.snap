// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`App component renders without error 1`] = `
<div
  className="App"
>
  <div
    className="wrapper"
  >
    <div
      className="app-temp"
    >
      <span
        className="app-temp-value"
      >
        26Â°C
      </span>
    </div>
    <div
      className="app-input"
    >
      <Dropdown
        additionLabel="Add "
        additionPosition="top"
        closeOnBlur={true}
        closeOnEscape={true}
        deburr={false}
        fluid={true}
        icon="dropdown"
        minCharacters={1}
        noResultsMessage="No results found."
        onChange={[Function]}
        openOnFocus={true}
        options={
          Array [
            Object {
              "key": "ru",
              "text": "Russia",
              "value": "ru",
            },
            Object {
              "key": "ua",
              "text": "Ukraine",
              "value": "ua",
            },
          ]
        }
        placeholder="Select Country..."
        renderLabel={[Function]}
        search={true}
        searchInput="text"
        selectOnBlur={true}
        selectOnNavigation={true}
        selection={true}
        wrapSelection={true}
      />
    </div>
    <div
      className="app-input"
    >
      <Search
        disabled={false}
        icon="search"
        input="text"
        loading={false}
        minCharacters={1}
        noResultsMessage="No results found."
        onResultSelect={[Function]}
        onSearchChange={[Function]}
        placeholder="Search city..."
        results={Array []}
        showNoResults={true}
        value=""
      />
    </div>
    <List
      selection={true}
    >
      <ListItem
        active={true}
        key="43"
        onClick={[Function]}
      >
        <ListContent
          floated="right"
        >
          <Button
            as="button"
            icon="delete"
            onClick={[Function]}
            size="mini"
          />
        </ListContent>
        <ListIcon
          name="marker"
          size="large"
          verticalAlign="middle"
        />
        <ListContent
          verticalAlign="middle"
        >
          <ListHeader>
            Saint-Peterspurg
          </ListHeader>
        </ListContent>
      </ListItem>
      <ListItem
        active={false}
        key="42"
        onClick={[Function]}
      >
        <ListContent
          floated="right"
        >
          <Button
            as="button"
            icon="delete"
            onClick={[Function]}
            size="mini"
          />
        </ListContent>
        <ListIcon
          name="marker"
          size="large"
          verticalAlign="middle"
        />
        <ListContent
          verticalAlign="middle"
        >
          <ListHeader>
            Kiev
          </ListHeader>
        </ListContent>
      </ListItem>
    </List>
  </div>
</div>
`;
